docs/docker-stack/build.rst

```diff
@@ -15,16 +15,126 @@
+and ending with the extra time needed to install those packages - which has to happen every time every
+container starts. The only viable way to deal with new dependencies and requirements in production is to
+build and use your own image. You should only use installing dependencies dynamically in case of
+"hobbyist" and "quick start" scenarios when you want to iterate quickly to try things out and later
+replace it with your own images.
+
+How to build your own image
+---------------------------
+
+There are several most-typical scenarios that you will encounter and here is a quick recipe on how to achieve
+your goal quickly. In order to understand details you can read further, but for the simple cases using
+typical tools here are the simple examples.
+
+In the simplest case building your image consists of those steps:
+
+1) Create your own ``Dockerfile`` (name it ``Dockerfile``) where you add:
+
+* information what your image should be based on (for example ``FROM: apache/airflow:latest-python3.8``
+
+* additional steps that should be executed in your image (typically in the form of ``RUN <command>``)
+
+2) Build your image. This can be done with ``docker`` CLI tools and examples below assume ``docker`` is used.
+   There are other tools like ``kaniko`` or ``podman`` that allow you to build the image, but ``docker`` is
+   so far the most popular and developer-friendly tool out there. Typical way of building the image looks
+   like follows (``my-custom-airflow-image-name`` is the custom name your image has). In case you use some
+   kind of registry where you will be using the image from, it is usually named in the form of
+   ``registry/image-name``. The name of the image has to be configured for the deployment method your
+   image will be deployed. This can be set for example as image name in the
+   `docker-compose file <running-airflow-in-docker>`_ or in the `Helm chart <helm-chart>`_.
+
+.. code-block:: shell
+
+   docker build . -f Dockerfile -t my-custom-airflow-image-name
+
+
+3) Once you build the image locally you have usually several options to make them available for your deployment:
+
+* For ``docker-compose`` deployment, that's all you need. The image is stored in docker engine cache
+  and docker compose will use it from there.
+
+* For some - development targeted - Kubernetes deployments you can load the images directly to
+  Kubernetes clusters. Clusters such as ``kind`` or ``minikube`` have dedicated ``load`` method to load the
+  images to the cluster.
+
+* Last but not least - you can push your image to a remote registry which is the most common way
+  of storing and exposing the images, and it is most portable way of publishing the image. Both
+  Docker-Compose and Kubernetes can make use of images exposed via registries.
+
+The most common scenarios where you want to build your own image are adding a new ``apt`` package,
```

Comment:
deb, not apt. Apt is the package manager, deb is the format.

Reply:
While technically you are right, similarly as in case of Docker images (which are technically Container images), most peoplpe have no idea they are using `deb` packages, but they know how to use `apt`. We are using the term 'apt dependencies' all-over our documentation and for the sake of consistency and better discoverability I prefer to leave it as 'apt dependencies' (even if this is technically not precise).  We are using debian buster as our base image so the `apt` term is more familiar to users of the image.

Similarly we use `PyPI` packages where technically those are `.whl` or '.sdist` packages. We also some time ago used `pip` packages but since some of our users use `poetry` or `pip tools` (even if it is not recommended) and 'PyPI` is familiar to most Python users we use the term `PyPI packages`.